rust/xaynet-server/src/storage/api.rs

```diff
@@ -0,0 +1,240 @@
+    async fn get_seed_dict(&mut self) -> StorageResult<Option<SeedDict>>;
+
+    /// Increments the mask score with the given [`MaskObject`].
+    /// The score of the given mask is incremented by `1`.
+    ///
+    /// # Behavior
+    ///
+    /// - If the mask score has been successfully incremented, return `Result::Ok(Vec<(MaskObject, u64)>)`.
+    /// - If the mask score could not be incremented due to a PET protocol error, return
+    ///   the corresponding `Result::Ok(MaskScoreIncr(Result::Err(MaskScoreIncrError)))`.
+    async fn incr_mask_score(
+        &mut self,
+        pk: &SumParticipantPublicKey,
+        mask: &MaskObject,
+    ) -> StorageResult<MaskScoreIncr>;
+
+    /// Gets the two masks with the highest score.
+    ///
+    /// # Behavior
+    ///
+    /// - If no masks exist, return `Result::Ok(Option::None)`.
+    /// - If only one mask exists, return this mask
+    ///   `Result::Ok(Option::Some(Vec<(MaskObject, u64)>))`.
+    /// - If two masks exist with the same score, return both
+    ///   `Result::Ok(Option::Some(Vec<(MaskObject, u64)>))`.
+    /// - If two masks exist with the different score, return
+    ///   both in descending order `Result::Ok(Option::Some(Vec<(MaskObject, u64)>))`.
+    async fn get_best_masks(&mut self) -> StorageResult<Option<Vec<(MaskObject, u64)>>>;
+
+    /// Gets the number of unique masks.
+    async fn get_number_of_unique_masks(&mut self) -> StorageResult<u64>;
+
+    /// Deletes all coordinator data. This includes the coordinator
+    /// state as well as the [`SumDict`], [`SeedDict`] and `mask` dictionary.
+    async fn delete_coordinator_data(&mut self) -> StorageResult<()>;
+
+    /// Deletes the [`SumDict`], [`SeedDict`] and `mask` dictionary.
+    async fn delete_dicts(&mut self) -> StorageResult<()>;
+
+    /// Sets the latest global model id.
+    ///
+    /// # Behavior
+    ///
+    /// - If no global model id has been set yet, set and return the new id `Result::Ok(String))`.
+    /// - If the global model id already exists, override and return the new id `Result::Ok(String))`.
+    async fn set_latest_global_model_id(&mut self, id: &str) -> StorageResult<()>;
+
+    /// Gets the latest global model id.
+    ///
+    /// # Behavior
+    ///
+    /// - If the global model id does not exist, return `Result::Ok(None)`.
+    /// - If the global model id exists, return `Result::Ok(String))`.
+    async fn get_latest_global_model_id(&mut self) -> StorageResult<Option<String>>;
+}
+
+#[async_trait]
+/// An abstract model storage.
+pub trait ModelStorage
+where
+    Self: Clone + Send + Sync + 'static,
+{
+    /// Sets a global model.
+    ///
+    /// # Behavior
+    ///
+    /// - If the global model already exists (has the same model id), return `Result::Err(StorageError))`.
+    /// - If the global model does not exist, set the model and return the corresponding id.
+    async fn set_global_model(&mut self, id: &str, global_model: &Model) -> StorageResult<()>;
```

Comment:
description doesn't match the return type, either

```suggestion
    /// - If the global model already exists (has the same model id), return `StorageResult::Err(StorageError)`.
    /// - If the global model does not exist, set the model and return `StorageResult::Ok(())`.
    async fn set_global_model(&mut self, id: &str, global_model: &Model) -> StorageResult<()>;
```

or

```suggestion
    /// - If the global model already exists (has the same model id), return `StorageResult::Err(StorageError)`.
    /// - If the global model does not exist, set the model and return the corresponding id `StorageResult::Ok(String)`.
    async fn set_global_model(&mut self, id: &str, global_model: &Model) -> StorageResult<String>;
```